src/storeController/index.ts

```diff
@@ -104,6 +107,15 @@ export default async function (
       }
     }
   }
+
+  async function upload (nodeVersion: string, pkg: {peripheralLocation: string, id: string}) {
+    const nodeMajor = nodeVersion.substring(0, nodeVersion.indexOf('.'))
+    const cachePath = path.join(store, 'side-effects-cache', nodeMajor, pkg.id)
+    if (await fs.exists(cachePath) || await fs.exists(`${cachePath}+stage`)) {
+      return
+    }
+    await copyPkg(pkg.peripheralLocation, cachePath, {filesResponse: { fromStore: true, filenames: [] }, force: true})
```

Comment:
Why always copying? We need to use the same import method that is used for importing modules from the store

Reply:
Because we don't have the `filesResponse.filenames` needed in case of `hardlink` or `auto`. We can fix it by using `cp -l` instead of `linkIndexedDir`, so that we can remove the `filesResponse.filenames` entirely. Otherwise we have to use copy.

Reply:
ok, let's have a separate story for this.